cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -791,7 +800,20 @@ public CloudQuota getQuota(String quotaName, boolean required) {
 
     @Override
     public List<ApplicationLog> getRecentLogs(String applicationName) {
-        throw new UnsupportedOperationException(MESSAGE_FEATURE_IS_NOT_YET_IMPLEMENTED);
+        UUID appGuid = getRequiredApplicationGuid(applicationName);
+        RecentLogsRequest request = RecentLogsRequest.builder()
+                                                     .applicationId(appGuid.toString())
+                                                     .build();
+        return dopplerClient.recentLogs(request)
+                            .map(Envelope::getLogMessage)
+                            .map(log -> new ApplicationLog(log.getApplicationId(),
+                                                           log.getMessage(),
+                                                           fromLogTimestamp(log.getTimestamp()),
+                                                           fromLogMessageType(log.getMessageType()),
+                                                           log.getSourceType(),
+                                                           log.getSourceInstance()))
+                            .collectSortedList(Comparator.comparing(ApplicationLog::getTimestamp))
```

Comment:
Do we need to sort them?

Reply:
Yes, the old controller client (v1.x) sorts them by timestamp

Reply:
If the response is already sorted, the sorted collector will not add negative overhead. If the response is not sorted, the method will be consistent with the ordered nature of the logs.